### PR TITLE
aws cloudprovider - unify and consolidate aws session creation

### PR DESCRIPTION
**Pull request by khizunov**
#### Which component this PR applies to?
cluster-autoscaler

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Unify and consolidate aws session creation (currently created in multiple places and with different configuration).
AWS SDK provider will configure and create the session according to provided configuration (e.g cloud config file)

#### Which issue(s) this PR fixes:
Non standard AWS regions (GovCloud, etc.) which uses different AWS api endpoints should use unified session configuration for all requests - currently GenerateEC2InstanceTypes doesn't take that into consideration.

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mwielgus** [comment]
Can you provide more information (preferably in the PR description) about what exactly has been done there and why do we need this?

**Comment by khizunov** [comment]
pinging @gjtempleton @feiskyer for approval

File: cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go [hunk 1]
```diff
@@ -83,7 +83,14 @@ func main() {
 	flag.Parse()
 	defer klog.Flush()
 
-	instanceTypes, err := aws.GenerateEC2InstanceTypes(*region)
+	sess, err := session.NewSession(&aws.Config{
```
Comment (gjtempleton) [inline cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go:86]: This file doesn't import session so `make generate` now fails with:

```
go generate ./cloudprovider/aws
# command-line-arguments
ec2_instance_types/gen.go:86:15: undefined: session
ec2_instance_types/gen.go:86:39: undefined: aws.Config
ec2_instance_types/gen.go:87:15: undefined: aws.String
cloudprovider/aws/aws_manager.go:17: running "go": exit status 2
```

**Review by gjtempleton — CHANGES_REQUESTED** [review]
Thanks for this, there's one thing needed to fix up with the instance generation functionality, but other than that, this looks good to go to me.

File: cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go (hunk 1 above)
Comment (khizunov) [reply cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go]: @gjtempleton The AWS Session doesn't work without region (maybe it worked some day I am not sure), I did some rework on gen.go if you think I should put it in different PR I can do that?

**Comment by gjtempleton** [comment]
Thanks for addressing. 

Looks good to me now.

/lgtm
/approve

**Pull request merged**